diff --git a/lang/gjs/src/gtk4/astalify.ts b/lang/gjs/src/gtk4/astalify.ts
index eb71bdb..a6750de 100644
--- a/lang/gjs/src/gtk4/astalify.ts
+++ b/lang/gjs/src/gtk4/astalify.ts
@@ -194,6 +194,16 @@ function setupControllers<T>(widget: Gtk.Widget, {
                 onButtonReleased?.(widget, event as Gdk.ButtonEvent)
             }
 
+            if (event.get_event_type() === Gdk.EventType.TOUCH_BEGIN) {
+                (event as any).get_button = () => 1;
+                onButtonPressed?.(widget, event as Gdk.ButtonEvent)
+            }
+
+            if (event.get_event_type() === Gdk.EventType.TOUCH_END) {
+                (event as any).get_button = () => 1;
+                onButtonReleased?.(widget, event as Gdk.ButtonEvent)
+            }
+
             onLegacy?.(widget, event)
         })
     }
